clients/src/main/java/org/apache/kafka/common/protocol/SendBuilder.java

```diff
@@ -122,6 +123,14 @@ public void writeVarlong(long i) {
         ByteUtils.writeVarlong(i, buffer);
     }
 
+    private void flushPendingSend() {
+        if (!buffers.isEmpty()) {
+            ByteBuffer[] byteBufferArray = buffers.toArray(new ByteBuffer[0]);
+            sends.add(new ByteBufferSend(destinationId, byteBufferArray));
```

Comment:
I just noticed that `ByteBufferSend` re-iterates over the byte buffers to compute the total size. We could compute the size while we accumulate them. I suppose that the number of buffers is usually small so it should not make a big difference.

Reply:
I was somewhat inclined to leave this as is, but finally I decided to do it since it might help (even if only little) with fetch overhead.